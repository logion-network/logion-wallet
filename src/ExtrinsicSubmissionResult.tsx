import Spinner from 'react-bootstrap/Spinner';
import { isSuccessful } from '@logion/client';

import { SignedTransaction } from './logion-chain/Signature';
import Alert from './common/Alert';

import './ExtrinsicSubmissionResult.css';

export interface Props {
    result?: SignedTransaction | null,
    error: any,
    successMessage?: string | JSX.Element,
    errorMessage?: string | JSX.Element,
    slim?: boolean,
}

export default function ExtrinsicSubmissionResult(props: Props) {

    const { result, error } = props;

    if(result === undefined && !error) {
        return null;
    }

    let alert = null;
    if(!error) {
        if(!result) {
            alert = (
                <Alert variant="polkadot" slim={ props.slim }>
                    <p>Submitting...</p>
                </Alert>
            );
        } else if(!isSuccessful(result || null)) {
            if(props.slim) {
                alert = (
                    <Alert variant="polkadot" slim={ props.slim }>
                        <p><Spinner animation="border" size="sm" /> {`Current status: ${result?.status.type}`}</p>
                    </Alert>
                );
            } else {
                alert = (
                    <Alert variant="polkadot" slim={ props.slim }>
                        <Spinner animation="border"/>
                        <p>{`Current status: ${result?.status.type}`}</p>
                    </Alert>
                );
            }
        } else {
            alert = (
                <Alert variant="polkadot" slim={ props.slim }>
                {
                    props.successMessage === undefined &&
                    <p>Submission successful.</p>
                }
                {
                    props.successMessage !== undefined && props.successMessage
                }
                </Alert>
            );
        }
    } else {
        alert = (
            <Alert variant="danger" slim={ props.slim }>
            {
                props.errorMessage === undefined &&
                <p>{`Submission failed: ${error.toString()}`}</p>
            }
            {
                props.errorMessage !== undefined && props.errorMessage
            }
            </Alert>
        );
    }

    return (
        <div className="ExtrinsicSubmissionResult">
            { alert }
        </div>
    );
}
