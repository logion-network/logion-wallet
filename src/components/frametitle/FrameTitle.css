.FrameTitle .icon-container + .text-container {
    margin-left: 20px;
}
