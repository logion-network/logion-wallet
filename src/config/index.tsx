import { EnvironmentString } from "@logion/client";

export interface ConfigType {
    environment: EnvironmentString | undefined,
    APP_NAME: string,
    directory: string,
    rpcEndpoints: string[],
    crossmintApiKey: string,
    logionClassification: string,
    creativeCommons: string,
}

export const DEFAULT_CONFIG: ConfigType = {
    environment: undefined,
    APP_NAME: "Logion Wallet",
    directory: "",
    rpcEndpoints: [],
    crossmintApiKey: "",
    logionClassification: "",
    creativeCommons: "",
};

export interface EnvConfigType extends Record<string, any> {

}

const configEnv: EnvConfigType = require(`./${process.env.NODE_ENV}.json`);

declare var CONFIG: any;

const config: ConfigType = { ...DEFAULT_CONFIG, ...configEnv, ...CONFIG };
export default config;
