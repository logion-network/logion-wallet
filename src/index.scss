@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400;600;700;800&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Lexend+Deca&display=swap');

$font-family-sans-serif: 'Montserrat', sans-serif;
$font-family-monospace: "Source Code Pro", monospace !default;
$grid-breakpoints: (
  xs: 0,
  sm: 576px,
  md: 768px,
  lg: 992px,
  xl: 1200px,
  xxl: 1400px,
  xxxl: 1600px
);
$container-max-widths: (
  sm: 540px,
  md: 720px,
  lg: 960px,
  xl: 1140px,
  xxl: 1320px,
  xxxl: 1400px
);


@import "~bootstrap/scss/bootstrap";

body, html {
    height: 100%;
    width: 100%;
}

html {
    font-size: 16px;
}

#root {
    height: 100%;
    width: 100%;
}

@media (max-width: 1350px) {
    html {
        font-size: 14px;
    }
}

.spinner-border.text-polkadot {
    color: #e6007a;
}
