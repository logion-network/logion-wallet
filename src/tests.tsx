import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };

export async function clickByName(name: string | RegExp | ((accessibleName: string, element: Element) => boolean)) {
    let button: HTMLElement;
    await waitFor(() => button = screen.getByRole("button", { name }));
    await userEvent.click(button!);
}

export async function typeByLabel(label: string, value: string) {
    let input: HTMLElement;
    await waitFor(() => input = screen.getByLabelText(label));
    await userEvent.type(input!, value);
}
