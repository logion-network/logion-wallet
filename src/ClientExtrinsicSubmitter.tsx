import { useEffect, useState } from 'react';

import ExtrinsicSubmissionResult from './ExtrinsicSubmissionResult';
import { useLogionChain } from './logion-chain';
import { Call, CallCallback } from './logion-chain/LogionChainContext';

export type { Call, CallCallback };

export interface Props {
    successMessage?: string | JSX.Element,
    call?: Call,
    onSuccess?: () => void,
    onError?: () => void,
    slim?: boolean,
}

export default function ClientExtrinsicSubmitter(props: Props) {
    const { extrinsicSubmissionState, submitCall, resetSubmissionState } = useLogionChain();
    const [ notified, setNotified ] = useState(false);

    useEffect(() => {
        if(props.call !== undefined && extrinsicSubmissionState.canSubmit()) {
            submitCall(props.call);
        }
    }, [ extrinsicSubmissionState, submitCall, props ]);

    useEffect(() => {
        if (extrinsicSubmissionState.isSuccessful() && !notified && props.onSuccess) {
            setNotified(true);
            resetSubmissionState();
            props.onSuccess();
        }
    }, [ extrinsicSubmissionState, props, notified, resetSubmissionState ]);

    useEffect(() => {
        if (extrinsicSubmissionState.isError() && !notified && props.onError) {
            setNotified(true);
            resetSubmissionState();
            props.onError();
        }
    }, [ extrinsicSubmissionState, props, notified, resetSubmissionState ]);

    if(props.call === undefined) {
        return null;
    }

    return (
        <ExtrinsicSubmissionResult
            result={extrinsicSubmissionState.result}
            error={extrinsicSubmissionState.error}
            successMessage={ props.successMessage }
            slim={ props.slim }
        />
    );
}
