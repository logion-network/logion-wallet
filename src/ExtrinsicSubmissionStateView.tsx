import { useLogionChain } from './logion-chain/LogionChainContext';
import ExtrinsicSubmissionResult from './ExtrinsicSubmissionResult';

export interface Props {
    successMessage?: string;
    slim?: boolean;
}

export default function ExtrinsicSubmissionStateView(props: Props) {
    const { extrinsicSubmissionState } = useLogionChain();

    if(extrinsicSubmissionState.submitted) {
        return (<ExtrinsicSubmissionResult
            error={ extrinsicSubmissionState.error }
            result={ extrinsicSubmissionState.result }
            successMessage={ props.successMessage }
            slim={ props.slim }
        />);
    } else {
        return null;
    }
}
