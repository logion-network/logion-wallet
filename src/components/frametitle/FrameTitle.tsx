import Icon from "src/common/Icon";

import "./FrameTitle.css";

export interface Props {
    iconId?: string;
    text: string;
}

export default function FrameTitle(props: Props) {
    return (
        <span className="FrameTitle">
            { props.iconId ? <span className="icon-container"><Icon icon={{id: props.iconId }} height="36px" /></span> : null }
            <span className="text-container">{ props.text }</span>
        </span>
    );
}
