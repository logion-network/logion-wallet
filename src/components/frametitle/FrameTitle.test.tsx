import { shallowRender } from "src/tests";

import FrameTitle from "./FrameTitle";

describe("FrameTitle", () => {

    it("renders without icon", () => {
        const element = shallowRender(<FrameTitle text="My Title" />);
        expect(element).toMatchSnapshot();
    });

    it("renders with icon", () => {
        const element = shallowRender(<FrameTitle text="My Title" iconId="ok" />);
        expect(element).toMatchSnapshot();
    });
});
