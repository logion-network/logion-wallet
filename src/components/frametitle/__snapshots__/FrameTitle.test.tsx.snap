// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrameTitle renders with icon 1`] = `
<span
  className="FrameTitle"
>
  <span
    className="icon-container"
  >
    <Icon
      height="36px"
      icon={
        Object {
          "id": "ok",
        }
      }
    />
  </span>
  <span
    className="text-container"
  >
    My Title
  </span>
</span>
`;

exports[`FrameTitle renders without icon 1`] = `
<span
  className="FrameTitle"
>
  <span
    className="text-container"
  >
    My Title
  </span>
</span>
`;
